core/src/main/java/org/mule/processor/IdempotentRedeliveryPolicy.java

```diff
@@ -285,10 +285,16 @@ private String getIdForEvent(MuleEvent event) throws Exception
     {
         if (useSecureHash)
         {
+            byte[] bytes;
             MuleMessage message = event.getMessage();
             Object payload = message.getPayload();
-            byte[] payloadBytes = (byte[]) objectToByteArray.transform(payload);
-            byte[] bytes = concat(payloadBytes, message.getUniqueId().getBytes());
+            if(Boolean.parseBoolean(System.getProperty("RollBackExceptionStrategy","false")))
```

Comment:
2 corrections: 
1. Static imports
2. Have this as a constant, since it won't change its value during the entire execution (reading purposes, but more importantly avoid going to check system property per each time the code is executed)